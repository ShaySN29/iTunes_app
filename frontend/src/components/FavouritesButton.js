import React from 'react';
// Imports from React Bootstrap
import { Button } from 'react-bootstrap';
// Importing the favourite icon
import favouriteIcon from '../images/favouriteIcon.png'

/* The FavouritesButton functional component displays the favourites button
  When the favourites button is clicked the state of the display favouritesList is set to true and the favourites list is shown */
function FavouritesButton({handleFavouriteButtonClick}) {
  return (
    <div className="favouriteBarDiv">
      <div className="iconDiv">
        <img src={favouriteIcon} height={40} className="mt-1 mb-3"/>
      </div>
      <div>
        <Button variant="success" className="mt-2 mb-3" onClick={e => handleFavouriteButtonClick()}>Favourites</Button>
      </div>
    </div>
  )
};

export default FavouritesButton;

/* ACKNOWLEDGEMENTS
Favourites icon used from: https://www.vecteezy.com/vector-art/379885-favorite-vector-icon */