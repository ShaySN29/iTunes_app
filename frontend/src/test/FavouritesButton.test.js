import React from "react";
import renderer from "react-test-renderer";
// Importing component to test render
import FavouritesButton from '../components/FavouritesButton.js';


it('renders correctly', () => {
    const tree = renderer.create(<FavouritesButton />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  }); 
