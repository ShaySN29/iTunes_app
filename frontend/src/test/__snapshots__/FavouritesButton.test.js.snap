// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="favouriteBarDiv"
>
  <div
    className="iconDiv"
  >
    <img
      className="mt-1 mb-3"
      height={40}
      src="favouriteIcon.png"
    />
  </div>
  <div>
    <button
      className="mt-2 mb-3 btn btn-success"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Favourites
    </button>
  </div>
</div>
`;
